### ISSUE TITLE
[CLI]: Tune Population-Based Bandits and WandbLoggerCallback Memory Leak

### ISSUE THREAD

**Issue by dkupsh** [wandb/wandb#5957]
### Describe the bug

Hello,

I'm currently experiencing a memory leak when using the WandbLoggerCallback with Ray Tune's Population-Based Bandit algorithm. The RAM Util is consistently increasing as running the experiment. The memory leak doesn't occur when running without wandb.

No error occurs in the command line, although ray reports that _WandbLoggingActor.run() fails. The error seems silent though (with only this vague "If you really want to do this, pass allow_val_change=True to config.update()"). From my understanding, wandb seems to create a new wandblogging instances for each trial, as it gives a wandb login message during each new trial.

### Environment

WandB version: 0.15.7

OS:

Python version: 3.8.16

Versions of relevant libraries:

Ray: 2.4.0

**Comment by rsanandres-wandb** [comment]
Hello @dkupsh !

Could you send the debug bundles of the run that is experiencing the memory leak? They should be located in the `wandb` folder in the same directory as where the script was run. The `wandb` folder has folders formatted as `run-DATETIME-ID` associated with a single run. Could you retrieve the debug.log and debug-internal.log files from one of these folders specifically from the run that is experiencing the memory leak?

**Comment by dkupsh** [comment]
Here is the [debug-internal.log](https://github.com/wandb/wandb/files/12264465/debug-internal.log). The debug.log file was empty.

**Comment by rsanandres-wandb** [comment]
Hello @dkupsh !

I looked into your debug logs and there does not seem to be anything necessarily wrong with the debug logs. However, you did mention that since each run is being created for each trial, this will increase memory and is to be expected if multiple `wandb` processes are being spun up. We also sync data asynchronously so older `wandb` processes may still be alive in order to upload metrics and files. 

Are you expecting multiple `wandb` processes to be spun up or you expecting one process (meaning one run) to be made during training?

**Comment by dkupsh** [comment]
Yes. So I only run `tune.fit` once, but a new `wandb.init() `seems to be spun up (based on action logs) during each time a new actor is created. On the ray console, it says that the `wandb` had an error when running (but there were no logs attached). 

Here's sortof an abridged version of the code:

```
pbt = PB2(
        time_attr="training_iteration",
        metric="episode_reward_mean",
        mode="max",
        perturbation_interval=100,
        ....
)
    
tuner = Tuner(
   'PPO',
   param_space=param_space,
   tune_config=tune.TuneConfig(
            num_samples=4,
            scheduler=pbt
    ),
    run_config=RunConfig(
       ...
      callbacks=[
                WandbLoggerCallback(
                   ....
          )
    )
) 

tuner.fit()
```

What I suspected was that the old `wandb` processes were still consuming system memory, thus causing the memory leak.

**Comment by rsanandres-wandb** [comment]
We don't have a direct integration Population-Based Bandits and after looking around it looks like you are right when it comes to `PB2` spinning up new `WandbLoggerCallback ` instances every-time it is called. I will let our engineers know that our integration currently doesn't support this and see what they have to say.

**Comment by sydholl** [comment]
WandB Internal User commented: 
rsanandres-wandb commented: 
Hello @dkupsh !

Could you send the debug bundles of the run that is experiencing the memory leak? They should be located in the `wandb` folder in the same directory as where the script was run. The `wandb` folder has folders formatted as `run-DATETIME-ID` associated with a single run. Could you retrieve the debug.log and debug-internal.log files from one of these folders specifically from the run that is experiencing the memory leak?

**Comment by sydholl** [comment]
WandB Internal User commented: 
dkupsh commented: 
Here is the [debug-internal.log](https://github.com/wandb/wandb/files/12264465/debug-internal.log). The debug.log file was empty.
